### PR TITLE
Create filesystem interface - Closes #865

### LINKED ISSUE #865: Create filesystem interface
### Description

Create a shared implementation of the filesystem data store.

Configuration:

- Local filesystem path with files (defaults to `./data/partials` and `./data/exports` if not found)
- Disk space limit in kB

### Acceptance Criteria

- Is able to write and read data incl. operations in the specs
- Unit tests

### Additional Information

Interface

```
module.exports = {
    init: (params) => {},
    write: async (filename, content) => {},
    read: async (filename) => {},
    delete: async (filename) => {},
    list: async (n, page) => {}, // non-production
    purge: async (days) => {},   // for cache maintenance
};
```

### PR DESCRIPTION
**Pull request by nagdahimanshu**
### What was the problem?
This PR resolves #865 

### How was it solved?
- [x] Is able to perform following operations:
  - [x] init
  - [x] read
  - [x] write
  - [x] delete
  - [x] list
  - [x] purge
- [x] Unit tests

### How was it tested?
Local + Jenkins

### CODE REVIEW

File: services/export/shared/helpers/file.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+			if (err) {
+				logger.error(err);
+				return reject(err);
+			}
+			return resolve(files.slice(page, page + count));
+		},
+	);
+});
+
+const purge = (dirPath, days = 1) => new Promise((resolve, reject) => {
```
Comment (sameersubudhi) [inline services/export/shared/helpers/file.js:99]: I recommend making the `days` param mandatory and adding a validation error.

File: services/export/shared/helpers/file.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+		await BluebirdPromise.map(
+			files,
+			async (file) => {
+				const stat = await getFileStats(path.join(dirPath, file));
+				const currentTime = new Date().getTime();
+				const expirationTime = new Date(stat.ctime).getTime() + getDaysInMilliseconds(days);
+				if (currentTime > expirationTime) await remove(path.join(dirPath, file));
+			},
+			{ concurrency: files.length },
+		);
```
Comment (sameersubudhi) [inline services/export/shared/helpers/file.js:114]: `remove` might throw errors. Please consider wrapping this inside of a `try..catch` and reject on failure.

**Review by sameersubudhi — CHANGES_REQUESTED** [review]
Please consider the following.

File: services/export/shared/helpers/file.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+					throw error;
+				}
+			},
+			{ concurrency: files.length },
+		);
+		return resolve();
+	});
+});
+
+const exists = filePath => fs.existsSync(filePath);
```
Comment (MichalTuleja) [inline services/export/shared/helpers/file.js:128]: `fs.existsSync()` could be updated to `fs.stat()` implementation.

**Review by MichalTuleja — CHANGES_REQUESTED** [review]

File: services/export/shared/helpers/file.js (hunk 3 above)
Comment (nagdahimanshu) [reply services/export/shared/helpers/file.js]: Done!

**Review by MichalTuleja — APPROVED** [review]
LGTM 👍

**Review by sameersubudhi — APPROVED** [review]
LGTM 👍

**Pull request merged**